yellowbrick/features/base.py

```diff
@@ -215,13 +225,58 @@ def fit(self, X, y=None, **kwargs):
         """
         super(DataVisualizer, self).fit(X, y, **kwargs)
 
-        # Store the classes for the legend if they're None.
-        if self.classes_ is None:
-            # TODO: Is this the most efficient method?
-            self.classes_ = [str(label) for label in np.unique(y)]
+        self._determine_target_color_type(y)
+
+        # Compute classes and colors if target type is discrete
+        if self._target_color_type == DISCRETE:
+            # Store the classes for the legend if they're None.
+            if self.classes_ is None:
+                # TODO: Is this the most efficient method?
+                self.classes_ = [str(label) for label in np.unique(y)]
+            
+# This fails in case of PCoords if subsampled target misses some values.
+            # Ensures that classes passed by user is equal to that in target
+#            if len(self.classes_)!=len(np.unique(y)):
+#                raise YellowbrickValueError("Number of unique target is not " 
+#                                            "equal to classes")
```

Comment:
Besides this comment being formatted weirdly - I'm not sure I understand what's going on here or why this is commented? Was this code here before? If so, can you look at git blame to see why it was commented out? I'd like to fix it if possible.

Reply:
Well, this was the functionality I added. It ensures that the number of classes is equal to the number of unique `target`. But it gave an error in case of ParallelCoord because it samples the target. Pcoords explicitly finds classes hence classes are equal to the unique unsampled target. Due to this, the number of classes becomes more than unique sampled y. So I seek suggestions from you on this. Should I keep or remove this? This may need updating pcoords.

Reply:
Was just looking at parallel coordinates, which looks like it may need some updates as a result of this PR; can you actually list all of the subclasses of `DataVisualizer` for me so that we can make sure that things work as expected? For example, we'll have to ensure that we do:

```
super(ParallelCoordinates, self).__init__(ax, target=DISCRETE) 
```

In `ParallelCoordinates` and `RadViz` since there isn't really a continuous target identification yet; and for those that do, we'll have to add the params of the `DataVisualizer` there. 

As for this functionality, I think doing a verification that the classes supplied by the user match whats in `y` is a good thing; perhaps instead of an exception, we could simply issue a warning?

Reply:
I updated with something like this:
```python
norm = matplotlib.colors.Normalize(self.range_[0], self.range_[1])
cmap = matplotlib.cm.get_cmap(self.colormap)
color_values = cmap([norm(yi) for yi in y])
self._colors = color_values
```
It calculates color for each point. So should I let it be like this or update to return just colormap.

Reply:
Following are the subclasses for DataVisualizer:

1. ParallelCoordinates
2. RadViz
3. contrib/missing/MissingDataVisualizer(It has only single target _type)
4. contrib/scatter(only discrete)

Reply:
Ok - that's actually great and not too many of them. Thanks for compiling that list! Right now none of these care about the target type, or they expect discrete.